### PR TITLE
[cmake] remove unnecessary include_directories in most ROOT tests

### PR DESCRIPTION
**Pull request by linev**
When using source bilds, all necessary source includes
will be provided via dependencies. When using installed root,
anyway ${CMAKE_BINARY_DIR}/include has to be used

Reduce usage of `find_package(ROOT)` method. 
If was misused to produce variables like `ROOT_asimage_FOUND` or `ROOT_vc_FOUND`
Now only kept at the place where it can be really used outside ROOT.
Plus two historical files in `test/` and `tutorials/` are mostly unchanged while too many such variables are used

### CODE REVIEW

**Comment by linev** [comment]
@phsft-bot build

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,9 @@
 cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
```
Comment (oshadura) [inline test/CMakeLists.txt:12]: cmake_minimum_required(VERSION 3.9) ?

File: test/periodic/CMakeLists.txt [hunk 2]
```diff
@@ -5,23 +5,25 @@
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
 cmake_minimum_required(VERSION 3.4 FATAL_ERROR)
```
Comment (oshadura) [inline test/periodic/CMakeLists.txt:7]: cmake_minimum_required(VERSION 3.9) ?

File: math/mathcore/test/CMakeLists.txt [hunk 3]
```diff
@@ -5,9 +5,6 @@
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
 project(mathcore-tests)
-find_package(ROOT REQUIRED)
```
Comment (oshadura) [inline math/mathcore/test/CMakeLists.txt:8]: Why here it was deleted while in `math/genetic/test/CMakeLists.txt ` if was organized as:
`if(NOT DEFINED ROOT_SOURCE_DIR)..endif()`

File: math/genetic/test/CMakeLists.txt [hunk 4]
```diff
@@ -4,19 +4,21 @@
 # For the licensing terms see $ROOTSYS/LICENSE.
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
+cmake_minimum_required(VERSION 3.9)
+
 project(genetic-tests)
```
Comment (oshadura) [inline math/genetic/test/CMakeLists.txt:9]: Lets move project(genetic-tests) as well inside if()..endif()?

File: test/CMakeLists.txt (hunk 1 above)
Comment (linev) [reply test/CMakeLists.txt:12]: I decide not to touch this file - just remove few `include_directory` commands.
`cmake_minimum_required` can be removed if we decide not to support standalone building of that directory

File: test/periodic/CMakeLists.txt (hunk 2 above)
Comment (linev) [reply test/periodic/CMakeLists.txt:7]: Ok

File: math/mathcore/test/CMakeLists.txt (hunk 3 above)
Comment (linev) [reply math/mathcore/test/CMakeLists.txt:8]: I comment in `Mattermost` - you also need googletest and it looks very ugly. 
Therefore now it is not supported at all to build mathcore test standalone

File: math/genetic/test/CMakeLists.txt (hunk 4 above)
Comment (linev) [reply math/genetic/test/CMakeLists.txt:9]: I will check

**Comment by linev** [comment]
> It is a bit confusing, it is possible to run this code as standalone still? I would actually remove 
> everywhere this possibility (e.g remove -> find_package(ROOT REQUIRED) and etc.)

I explained in Mattermost - `find_package(ROOT REQUIRED)` kept in two places while variables like `ROOT_vs_FOUND` or `ROOT_mathcore_FOUND` defined only after calling `ROOTConfigure.cmake`. One need to change a lot in `test/CMakeLists.txt` and `tutorials/CMakeLists.txt`

**Comment by oshadura** [comment]
Ok, so you mean it could or should be done in other iteration after landing this PR?

**Comment by linev** [comment]
> Ok, so you mean it could or should be done in other iteration after landing this PR?

Yes, there are many options which should be touched and I not always sure if to 1-to-1 relation

**Comment by oshadura** [comment]
I will be thankful if you could update minimal version of CMake everywhere in these files (it is too much noticeable).

**Comment by linev** [comment]
@oshadura I adjust cmake version, lets wait for jenkins. Also please check #5573

**Comment by linev** [comment]
@phsft-bot build

**Comment by linev** [comment]
Finally all tests passed!

**Pull request merged**